api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -136,15 +136,27 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
+	allErrs = append(allErrs, m.ValidateCommonFields(specPath)...)
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(GroupVersion.WithKind("MachineHealthCheck").GroupKind(), m.Name, allErrs)
+}
+
+// ValidateCommonFields validates UnhealthyConditions NodeStartupTimeout, MaxUnhealthy, and RemediationTemplate of the MHC.
+// It is used to help validate other types which define MachineHealthChecks such as MachineHealthCheckClass and MachineHealthCheckTopology.
```

Comment:
```suggestion
// It is also used to help validate other types which define MachineHealthChecks such as MachineHealthCheckClass and MachineHealthCheckTopology.
```
nit (given it's also used for MHC itself)